buildlib/pr/mad_tests.yml

```diff
@@ -0,0 +1,114 @@
+jobs:
+  - job: SetupServer
+    pool:
+      name: MLNX
+      demands: mad_server
+    workspace:
+      clean: outputs
+    steps:
+      - checkout: self
+        clean: true
+        fetchDepth: 100
+        retryCountOnTaskFailure: 5
+      - bash: |
+          source ./contrib/test_mad.sh
+          setup
+          build_ucx
+        displayName: Setup Server
+
+  - job: SetupClient
+    pool:
+      name: MLNX
+      demands: mad_client
+    workspace:
+      clean: outputs
+    steps:
+      - checkout: self
+        clean: true
+        fetchDepth: 100
+        retryCountOnTaskFailure: 5
+      - bash: |
+          source ./contrib/test_mad.sh
+          setup
+          build_ucx
+        displayName: Setup Client
+
+  - job: ServerUp
+    dependsOn: SetupServer
+    displayName: Server Up
+    pool:
+      name: MLNX
+      demands: mad_server
+    steps:
+      - checkout: none
+      - bash: |
+          source ./contrib/test_mad.sh
+          run_mad_server
+        displayName: Server Up
+
+      - task: Bash@3
+        name: Set_Vars
+        inputs:
+          targetType: "inline"
+          script: |
+            source ./contrib/test_mad.sh
+            set_vars
+        displayName: Set Vars
+
+  - job: TestLid
+    dependsOn:
+      - ServerUp
+      - SetupClient
+    displayName: Test Lid
+    timeoutInMinutes: 10
+    pool:
+      name: MLNX
+      demands: mad_client
+    variables:
+      LID: $[ dependencies.ServerUp.outputs['Set_Vars.LID'] ]
+    steps:
+      - checkout: none
+      - bash: $(Build.SourcesDirectory)/install/bin/ucx_perftest -t tag_bw -e -K $(HCA) -e lid:$(LID)
```

Comment:
tbh, I preferred previous version where commands were coming from sourced bash, easier to run/test on dev machines.

Reply:
Your wish has been granted.